addons/html_editor/static/src/editor/core/utils.js

```diff
@@ -30,3 +30,133 @@ export function parseHTML(document, html) {
     fragment.replaceChildren(...parsedDocument.body.childNodes);
     return fragment;
 }
+
+const blockTagNames = [
+    "ADDRESS",
+    "ARTICLE",
+    "ASIDE",
+    "BLOCKQUOTE",
+    "DETAILS",
+    "DIALOG",
+    "DD",
+    "DIV",
+    "DL",
+    "DT",
+    "FIELDSET",
+    "FIGCAPTION",
+    "FIGURE",
+    "FOOTER",
+    "FORM",
+    "H1",
+    "H2",
+    "H3",
+    "H4",
+    "H5",
+    "H6",
+    "HEADER",
+    "HGROUP",
+    "HR",
+    "LI",
+    "MAIN",
+    "NAV",
+    "OL",
+    "P",
+    "PRE",
+    "SECTION",
+    "TABLE",
+    "UL",
+    // The following elements are not in the W3C list, for some reason.
+    "SELECT",
+    "OPTION",
+    "TR",
+    "TD",
+    "TBODY",
+    "THEAD",
+    "TH",
+];
+
+const computedStyles = new WeakMap();
+
+/**
+ * Return true if the given node is a block-level element, false otherwise.
+ *
+ * @param node
+ */
+export function isBlock(node) {
+    if (!node || node.nodeType !== Node.ELEMENT_NODE) {
+        return false;
+    }
+    const tagName = node.nodeName.toUpperCase();
+    // Every custom jw-* node will be considered as blocks.
+    // @todo @phoenix check if this should be removed
+    // if (
+    //     tagName.startsWith("JW-") ||
```

Comment:
This would should be removed for sure (old Jabberwok stuff).
Not sure about the T tag.